node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let empty_input = vec![];
+
+        let result = subject.insert_new_records(&empty_input);
+
+        assert_eq!(result, Err(FailedPayableDaoError::EmptyInput));
+    }
+
+    #[test]
+    fn insert_new_records_throws_error_when_two_txs_with_same_hash_are_present_in_the_input() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "insert_new_records_throws_error_when_two_txs_with_same_hash_are_present_in_the_input",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let hash = make_tx_hash(123);
+        let tx1 = FailedTxBuilder::default().hash(hash).build();
+        let tx2 = FailedTxBuilder::default().hash(hash).checked(1).build();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+
+        let result = subject.insert_new_records(&vec![tx1, tx2]);
+
+        assert_eq!(
+            result,
+            Err(FailedPayableDaoError::InvalidInput(
+                "Duplicate hashes found in the input".to_string()
+            ))
+        );
+    }
+
+    #[test]
+    fn insert_new_records_throws_error_when_input_tx_hash_is_already_present_in_the_db() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "insert_new_records_throws_error_when_input_tx_hash_is_already_present_in_the_db",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let hash = make_tx_hash(123);
+        let tx1 = FailedTxBuilder::default().hash(hash).build();
+        let tx2 = FailedTxBuilder::default().hash(hash).checked(1).build();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let initial_insertion_result = subject.insert_new_records(&vec![tx1]);
+
+        let result = subject.insert_new_records(&vec![tx2]);
+
+        assert_eq!(initial_insertion_result, Ok(()));
+        assert_eq!(
+            result,
+            Err(FailedPayableDaoError::InvalidInput(
+                "Input hash is already present in the database".to_string()
+            ))
+        );
+    }
+
+    #[test]
+    fn insert_new_records_returns_err_if_partially_executed() {
+        let setup_conn = Connection::open_in_memory().unwrap();
+        setup_conn
+            .execute("CREATE TABLE example (id integer)", [])
+            .unwrap();
+        let get_tx_identifiers_stmt = setup_conn.prepare("SELECT id FROM example").unwrap();
+        let faulty_insert_stmt = { setup_conn.prepare("SELECT id FROM example").unwrap() };
+        let wrapped_conn = ConnectionWrapperMock::default()
+            .prepare_result(Ok(get_tx_identifiers_stmt))
+            .prepare_result(Ok(faulty_insert_stmt));
```

Comment:
Here you are preparing for two transactions but eventually the test performs over just one.

Reply:
It'll require both of them. Since, we need to check for unique hashes.